lms/static/scripts/file_picker_v2/components/LMSFilePicker.js

```diff
@@ -25,19 +27,20 @@ export default function LMSFilePicker({
   onCancel,
   onSelectFile,
 }) {
-  /** The array of files returned by a call to `listFiles`. */
-  const [files, setFiles] = useState([]);
+  // The main state of the dialog and associated data.
+  const [{ state, files, error }, setState] = useState({
+    // The current state of the dialog, one of:
+    // "fetching", "fetched", "authorizing" or "error".
+    state: isAuthorized ? 'fetching' : 'authorizing',
 
-  /** Set to `true` if the list of files is being fetched. */
-  const [isLoading, setLoading] = useState(true);
+    // List of fetched files. Set when state is "fetched".
+    files: null,
 
-  /**
-   * `true` if we are waiting for the user to authorize the app's access
-   * to files in the LMS.
-   */
-  const [isAuthorizing, setAuthorizing] = useState(!isAuthorized);
+    // Fetch error details. Set when state is "error".
+    error: null,
+  });
```

Comment:
The purpose of refactoring several `useState` calls into one here is to represent the state of the dialog in such a way that it cannot be in an invalid combination of states (eg. "fetching" and "authorizing" at the same time, or "authorizing" but with a non-empty file list)